sdks/java/extensions/euphoria/euphoria-core/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/translate/coder/RegisterCoders.java

```diff
@@ -0,0 +1,166 @@
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+/**
+ * Convenient way of registering Beam {@link Coder} to given {@link Pipeline} or {@link BeamFlow}.
+ */
+public class RegisterCoders extends CoderProvider {
+
+  private final Map<TypeDescriptor, Coder<?>> typeToCoder;
+  private final Map<Class<?>, Coder<?>> classToCoder;
+
+  private RegisterCoders(
+      Map<TypeDescriptor, Coder<?>> typeToCoder,
+      Map<Class<?>, Coder<?>> classToCoder) {
+    this.typeToCoder = typeToCoder;
+    this.classToCoder = classToCoder;
+  }
+
+  public static RegisterBuilder to(Pipeline pipeline) {
+    return new Builder(Objects.requireNonNull(pipeline));
+  }
+
+  public static RegisterBuilder to(BeamFlow flow) {
+    return to(Objects.requireNonNull(flow).getPipeline());
+  }
+
+  @Override
+  public <T> Coder<T> coderFor(TypeDescriptor<T> typeDescriptor,
+      List<? extends Coder<?>> componentCoders) throws CannotProvideCoderException {
+
+    // try to obtain most specific coder by type descriptor
+    Coder<?> coder = typeToCoder.get(typeDescriptor);
+
+    // second try, obtain coder by raw encoding type
+    if (coder == null) {
+      Class<? super T> rawType = typeDescriptor.getRawType();
+      coder = classToCoder.get(rawType);
+    }
+
+    if (coder == null) {
+      throw new CannotProvideCoderException(String.format(
+          "No coder for given type descriptor '%s' found.", typeDescriptor));
+    }
+
+    @SuppressWarnings("unchecked")
+    Coder<T> castedCoder = (Coder<T>) coder;
+
+    return castedCoder;
+  }
+
+  // ----------------------------- builder chain
+
+  /**
+   * Builder which defines all the registration methods.
+   */
+  public interface RegisterBuilder {
+
+    /**
+     * Registers custom {@link Coder} for given parametrized {@link TypeDescriptor}.
+     *
+     * @param type type to register coder for
+     * @param coder coder to register
+     * @param <T> type of elements encoded by given {@code coder}
+     * @return {@link RegisterBuilder} to allow for more coders registration.
+     */
+    <T> RegisterBuilder regCustomTypedCoder(TypeDescriptor<T> type, Coder<T> coder);
+
+    /**
+     * Registers custom {@link Coder} for given raw {@link Class type}.
+     *
+     * @param clazz type to register coder for
+     * @param coder coder to register
+     * @param <T> type of elements encoded by given {@code coder}
+     * @return {@link RegisterBuilder} to allow for more coders registration.
+     */
+    <T> RegisterBuilder regCustomRawTypeCoder(Class<T> clazz, Coder<T> coder);
+
+    /**
+     * Registers new {@link ClassAwareKryoCoder} for given raw {@link Class type}.
+     *
+     * @param clazz type to register coder for
+     * @param <T> type of elements encoded by given {@code coder}
+     * @return {@link RegisterBuilder} to allow for more coders registration.
+     */
+    <T> RegisterBuilder regKryoRawTypeCoder(Class<T> clazz);
```

Comment:
`registerKryoCoder`, we should probably also have `Class<T> clazz, Serializer<T> serializer` variant for kryo

Reply:
Shure, i will address it in  [BEAM-4705](https://issues.apache.org/jira/browse/BEAM-4705).